### ISSUE TITLE
Abstractify everything

### ISSUE THREAD

**Issue by britram** [taps-api/drafts#176]
Make sure all the examples in the API and interface drafts use constants agnostic to argument type and syntax. From Dave Thaler:

On 2 May 2018, at 03:07, Dave Thaler <dthaler@microsoft.com> wrote:

> I skimmed the TAPS interface draft.  I like that it aims to be an abstract, not concrete API.
> But then it contains concrete examples that aren’t explained or, in my view, useful.  E.g.:
 
```
SecurityParameters.AddSupportedGroup(22) // secp256k1
SecurityParameters.AddCiphersuite(0xCCA9) // TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 SecurityParameters.AddSignatureAlgorithm(7) // ed25519
```

> Is less useful than a more abstract:

``` 
SecurityParameters.AddSupportedGroup(secp256k1)
SecurityParameters.AddCiphersuite(TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256)
SecurityParameters.AddSignatureAlgorithm(ed25519)
```
> The latter is agnostic as to the argument type, and the comment syntax.

**Comment by britram** [comment]
#197 will fix this when it lands

**Comment by chris-wood** [comment]
Addressed by #197.
